sympy/integrals/tests/test_manual.py

```diff
@@ -181,19 +181,26 @@ def test_manualintegrate_inversetrig():
     assert manualintegrate(1/sqrt(x**2 - 4), x) == log(2*x + 2*sqrt(x**2 - 4))
     assert manualintegrate(1/sqrt(4*x**2 - 4), x) == log(8*x + 4*sqrt(4*x**2 - 4))/2
     assert manualintegrate(1/sqrt(9*x**2 - 1), x) == log(18*x + 6*sqrt(9*x**2 - 1))/3
-    assert manualintegrate(1/sqrt(ra*x**2 - 4), x) == log(2*sqrt(ra)*sqrt(ra*x**2 - 4) + 2*ra*x)/sqrt(ra)
+    assert manualintegrate(1/sqrt(ra*x**2 - 4), x) == \
+           Piecewise((log(2*sqrt(ra)*sqrt(ra*x**2 - 4) + 2*ra*x)/sqrt(ra), Ne(ra, 0)), (-I*x/2, True))
     assert manualintegrate(1/sqrt(-ra + 4*x**2), x) == \
         Piecewise((asinh(2*x*sqrt(-1/ra))/2, ra < 0), (log(8*x + 4*sqrt(-ra + 4*x**2))/2, True))
 
     # From https://www.wikiwand.com/en/List_of_integrals_of_inverse_trigonometric_functions
     # asin
     assert manualintegrate(asin(x), x) == x*asin(x) + sqrt(1 - x**2)
     assert manualintegrate(asin(a*x), x) == Piecewise(((a*x*asin(a*x) + sqrt(-a**2*x**2 + 1))/a, Ne(a, 0)), (0, True))
-    assert manualintegrate(x*asin(a*x), x) == -a*Integral(x**2/sqrt(-a**2*x**2 + 1), x)/2 + x**2*asin(a*x)/2
+    assert manualintegrate(x*asin(a*x), x) == \
+           -a*Piecewise((-x*sqrt(-a**2*x**2 + 1)/(2*a**2) +
+                         log(-2*a**2*x + 2*sqrt(-a**2)*sqrt(-a**2*x**2 + 1))/(2*a**2*sqrt(-a**2)), Ne(a**2, 0)),
+                        (x**3/3, True))/2 + x**2*asin(a*x)/2
     # acos
     assert manualintegrate(acos(x), x) == x*acos(x) - sqrt(1 - x**2)
     assert manualintegrate(acos(a*x), x) == Piecewise(((a*x*acos(a*x) - sqrt(-a**2*x**2 + 1))/a, Ne(a, 0)), (pi*x/2, True))
-    assert manualintegrate(x*acos(a*x), x) == a*Integral(x**2/sqrt(-a**2*x**2 + 1), x)/2 + x**2*acos(a*x)/2
+    assert manualintegrate(x*acos(a*x), x) == \
+           a*Piecewise((-x*sqrt(-a**2*x**2 + 1)/(2*a**2) +
+                        log(-2*a**2*x + 2*sqrt(-a**2)*sqrt(-a**2*x**2 + 1))/(2*a**2*sqrt(-a**2)), Ne(a**2, 0)),
+                       (x**3/3, True))/2 + x**2*acos(a*x)/2
```

Comment:
Is the second case here correct?

I would have though that if `Ne(a**2, 0)` is False then we have `a==0` in which case:
```python
In [45]: x*acos(0*x)
Out[45]: 
π⋅x
───
 2 

In [46]: integrate(_, x)
Out[46]: 
   2
π⋅x 
────
 4  
```
The final term seems to match that but where does the `x**3/3` come from?

Reply:
I must be missing something:
```python
In [57]: x*acos(0)
Out[57]: 
π⋅x
───
 2 

In [58]: diff(x**3/3 + x**2/2*acos(0), x)
Out[58]: 
 2   π⋅x
x  + ───
      2 
```
The `x**3` seems to give a redundant term after differentiating.

Reply:
There is a `a` that multiplies `Piecewise`, so `0*x**3/3 = 0`